### DISCUSSION TITLE
How can I play an audio file that is still being written?

### DISCUSSION THREAD

**Discussion by yarslvd** [expo/expo#32500] [Archived: Questions]
I want to play audio which is streamed by the server in chunks. For that I have created a function which appends a new chunk of data in the specific file.
```
export const writeBase64ToFile = async (base64Data: string, path: string) => {
  const stream = await RNFetchBlob.fs.writeStream(path, 'base64', true);

  try {
    stream.write(base64Data);
  } catch (err) {
    console.error('Error writing to file:', err);
  } finally {
    stream.close();
  }
};
```

I am starting playing my audio only after first chunk has been received and written into file:
```
    es.addEventListener('message', async (event: VoiceEvent) => {
      const audioChunk = event.data;

      if (audioChunk) {
        await writeBase64ToFile(audioChunk, AUDIO_FILE_PATH);

        console.log('play');
        if (!isPlaying) {
          console.log('play inside');
          isPlaying = true;
          await playAudio(AUDIO_FILE_PATH);
        }
      }
    });
```

The audio written in file correctly, so I can open it and play with audio player. The audio doesn't start playing if the file was just created. Here is the playAudio function:
```
export const playAudio = async (path: string, callback?: () => void) => {
  try {
    const filePath = path.startsWith('file://') ? path : `file://${path}`;
    const {sound} = await Audio.Sound.createAsync(
      {uri: filePath},
      {shouldPlay: true},
    );

    sound.setOnPlaybackStatusUpdate(async status => {
      //@ts-expect-error Handling full audio play
      if (status.didJustFinish) {
        callback?.();
        await clearAudioMessage(path);
        console.log('cleared');
      }
    });
  } catch (err) {
    console.error(err);
  }
};
```

I tried starting audio playback after receiving the 3 chunk of data, which successfully begins playback, but it interrupts after a few seconds, even though the audio file is saved correctly. How can I resolve this issue? I use react native cli with expo modules.

**Comment by s4rath07** [comment]
Did you get any solution??

**Comment by zwily** [comment]
I'm currently looking into how to do the same thing...

**Reply by yarslvd** [reply]
> I'm currently looking into how to do the same thing...

I guess the only solution is to write native module
